ast/src/parsed/pattern.rs

```diff
@@ -0,0 +1,411 @@
+use powdr_number::BigInt;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use std::iter::{empty, once};
+
+use super::{asm::SymbolPath, visitor::Children};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PatternTuple {
```

Comment:
Can you document a little bit what this is beyond just a vector of patterns? Also please make clear that this is not part of the AST.